.github/workflows/admin.yml

```diff
@@ -47,31 +34,29 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Set up Go release
-        if: env.GO_VERSION != 'tip'
         uses: actions/setup-go@v4
         with:
-          go-version: ${{ env.GO_VERSION }}
+          go-version-file: ${{ github.workspace }}/go.mod
+          cache: false
 
-      - name: Set up Go tip
-        if: env.GO_VERSION == 'tip'
-        run: |
-          git clone --depth=1 https://go.googlesource.com/go $HOME/gotip
-          cd $HOME/gotip/src
-          ./make.bash
-          echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
-          echo "$HOME/gotip/bin" >> $GITHUB_PATH
-          echo "$GOBIN" >> $GITHUB_PATH
+      - name: Enable Go build cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-build-${{ github.ref }}-${{ hashFiles('**') }}
+          restore-keys: |
+            ${{ runner.os }}-go-build-${{ github.ref }}-
+            ${{ runner.os }}-go-build-
 
       - name: Enable Go modules cache
         uses: actions/cache@v3
         with:
           path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ env.GO_VERSION }}-modules-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-${{ env.GO_VERSION }}-modules-
+          key: ${{ runner.os }}-go-modules-${{ hashFiles('**/go.sum') }}
```

Comment:
This is why we can't use embedded in `setup-go` action cache. Our cache uses all `go.mod` files in the project, while setup-go can use only one file. However this feature may be implemented soon: https://github.com/actions/setup-go/issues/371